test_data/postgres_structured_report.json

```diff
@@ -0,0 +1,154 @@
+                   "Column 'd': type circle is mapped to string(max). No appropriate Spanner type"
+                ]
+             }
+          ]
+       },
+       {
+          "srcTableName":"default_value",
+          "spTableName":"default_value",
+          "schemaReport":{
+             "rating":"POOR",
+             "pkMissing":false,
+             "warnings":1,
+             "totalColumns":2
+          },
+          "dataReport":{
+             "rating":"NONE",
+             "badRows":0,
+             "totalRows":0,
+             "dryRun":false
+          },
+          "warnings":[
+             {
+                "warningType":"Warning",
+                "warningList":[
+                   "Some columns have default values which Spanner does not support e.g. column 'b'"
+                ]
+             }
+          ]
+       },
+       {
+          "srcTableName":"excellent_schema",
+          "spTableName":"excellent_schema",
+          "schemaReport":{
+             "rating":"EXCELLENT",
+             "pkMissing":false,
+             "warnings":0,
+             "totalColumns":2
+          },
+          "dataReport":{
+             "rating":"NONE",
+             "badRows":0,
+             "totalRows":0,
+             "dryRun":false
+          },
+          "warnings":null
+       },
+       {
+          "srcTableName":"foreign_key",
+          "spTableName":"foreign_key",
+          "schemaReport":{
+             "rating":"EXCELLENT",
+             "pkMissing":false,
+             "warnings":0,
+             "totalColumns":2
+          },
+          "dataReport":{
+             "rating":"NONE",
+             "badRows":0,
+             "totalRows":0,
+             "dryRun":false
+          },
+          "warnings":null
+       },
+       {
+          "srcTableName":"no_pk",
+          "spTableName":"no_pk",
+          "schemaReport":{
+             "rating":"POOR",
+             "pkMissing":true,
+             "warnings":1,
+             "totalColumns":3
+          },
+          "dataReport":{
+             "rating":"POOR",
+             "badRows":2000,
+             "totalRows":5000,
+             "dryRun":false
+          },
+          "warnings":[
+             {
+                "warningType":"Warnings",
+                "warningList":[
+                   "Column 'synth_id' was added because this table didn't have a primary key. Spanner requires a primary key for every table",
+                   "Some columns will consume more storage in Spanner e.g. for column 'b', source DB type int4 is mapped to Spanner data type int64"
+                ]
+             }
+          ]
+       }
+    ],
+    "unexpectedConditions":{
+       "Reparsed":0,
+       "unexpectedConditions":[
+          {
+             "count":5,
+             "condition":"Testing unexpected messages"
+          }
+       ]
+    },
+    "SchemaOnly":false
```

Comment:
why do we need a schemaOnly attribute if we have migration type?

Reply:
Removed `SchemaOnly` from JSON serialisation.